test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -148,6 +149,96 @@ func (v *volumeModifyTestSuite) DefineTests(driver storageframework.TestDriver,
 		framework.ExpectNoError(errors.NewAggregate(errs), "While cleaning up")
 	}
 
+	ginkgo.It("shouldn't bind the existing pv when the pvc doesn't contain a vac but a pvc require it", func(ctx context.Context) {
+		init(ctx, false /* volume created without VAC */)
+		ginkgo.DeferCleanup(cleanup)
+
+		ginkgo.By("Creating a PV without VAC")
+		pv := l.resource.Pv
+		pv1 := &v1.PersistentVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   pv.Name + "-1",
+				Labels: pv.Labels,
+			},
+			Spec: pv.Spec,
+		}
+		pv1.Spec.ClaimRef = nil
+		pv1.Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimRetain
+
+		_, err := f.ClientSet.CoreV1().PersistentVolumes().Create(ctx, pv1, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "While creating PV")
+
+		ginkgo.By("Creating a PVC with VAC")
+		pvc := l.resource.Pvc
+		pvc1 := &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pvc.Name + "-1",
+				Namespace: pvc.Namespace,
+				Labels:    pvc.Labels,
+			},
+			Spec: pvc.Spec,
+		}
+		pvc1.Spec.VolumeName = ""
+		pvc1.Spec.VolumeAttributesClassName = &l.vac.Name
+
+		_, err = f.ClientSet.CoreV1().PersistentVolumeClaims(pvc1.Namespace).Create(ctx, pvc1, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "While creating PVC")
+
+		ginkgo.By("Waiting for PVC to be in bound phase but not bound to that PV")
```

Comment:
This test case may not be very determinsitic because it could depend on the ordering that pv controller iterates through the list of PVs. I think checking that it doesn't bind to anything is probably better. You can potentially check for events to speed up the test (but not completely rely on events)